src/raft.rs

```diff
@@ -2483,9 +2522,90 @@ impl<T: Storage> Raft<T> {
             to_send.reject = true;
             to_send.reject_hint = hint_index;
             to_send.log_term = hint_term.unwrap();
+            success = false;
         }
         to_send.set_commit(self.raft_log.committed);
         self.r.send(to_send, &mut self.msgs);
+        success
+    }
+
+    // TODO: revoke pub when there is a better way to test.
+    /// For a given message, append the entries to the log.
+    pub fn handle_append_entries(&mut self, m: &Message) {
+        self.try_append_entries(m);
+    }
+
+    // For a broadcast, append entries to onw log and forward MsgAppend to other dest.
+    fn handle_group_broadcast(&mut self, m: &Message) {
+        if self.try_append_entries(m) {
+            // If the agent fails to append entries from the leader,
+            // the agent cannot forward MsgAppend.
+            for forward in m.get_forwards() {
+                // Fetch log entries from the forward.index to the last index of log.
+                if self
+                    .raft_log
+                    .match_term(forward.get_index(), forward.get_log_term())
+                {
+                    let ents = self.raft_log.entries(
+                        forward.get_index() + 1,
+                        self.max_msg_size,
+                        GetEntriesContext(GetEntriesFor::SendAppend {
+                            to: forward.get_to(),
+                            term: m.term,
+                            aggressively: false,
+                        }),
+                    );
+
+                    match ents {
+                        Ok(ents) => {
+                            let mut m_append = Message::default();
+                            m_append.to = forward.get_to();
+                            m_append.from = m.get_from();
+                            m_append.set_msg_type(MessageType::MsgAppend);
+                            m_append.index = forward.get_index();
+                            m_append.log_term = forward.get_log_term();
+                            m_append.set_entries(ents.into());
+                            m_append.commit = m.get_commit();
+                            m_append.commit_term = m.get_commit_term();
+                            self.r.send(m_append, &mut self.msgs);
+                        }
+                        Err(_) => {
```

Comment:
consider temporary unavailable

Reply:
Is the processing like the way in `fn send_append_aggressively`, looping for fetching until it success?

Reply:
nope, call it again in `on_entries_fetched`

Reply:
Async fetch for forwarding is committed. PTAL.